src/plugins/ucx/ucx_backend.cpp

```diff
@@ -592,35 +626,37 @@ nixl_status_t nixlUcxEngine::connect(const std::string &remote_agent) {
         return NIXL_ERR_NOT_FOUND;
     }
 
-    nixlUcxConnection &conn = remoteConnMap[remote_agent];
-
     hdr.op = CONN_CHECK;
     //agent names should never be long enough to need RNDV
     flags |= UCP_AM_SEND_FLAG_EAGER;
 
-    ret = uw->sendAm(conn.getEp(), CONN_CHECK,
-                     &hdr, sizeof(struct nixl_ucx_am_hdr),
-                     (void*) localAgent.data(), localAgent.size(),
-                     flags, req);
-
-    if(ret < 0) {
-        return ret;
+    bool error = false;
+    nixl_status_t ret = NIXL_SUCCESS;
+    std::vector<nixlUcxReq> reqs;
+    for (size_t i = 0; i < uws.size(); i++) {
+        reqs.emplace_back();
+        ret = search->second->getEp(i)->sendAm(CONN_CHECK,
+                                               &hdr, sizeof(struct nixl_ucx_am_hdr),
+                                               (void*) localAgent.data(), localAgent.size(),
+                                               flags, reqs.back());
+        if(ret < 0) {
+            error = true;
+            break;
+        }
     }
 
     //wait for AM to send
-    while(ret == NIXL_IN_PROG){
-        ret = uw->test(req);
-    }
+    for (size_t i = 0; i < reqs.size(); i++)
```

Comment:
shall we initialize ret before this loop?

Reply:
done